internal/controllers/gateway/gateway_controller.go

```diff
@@ -192,6 +236,19 @@ func (r *GatewayReconciler) isGatewayService(obj client.Object) bool {
 	return fmt.Sprintf("%s/%s", obj.GetNamespace(), obj.GetName()) == r.PublishService
 }
 
+func hasGatewayFrom(obj client.Object) bool {
```

Comment:
I believe the name should reflect the fact that we expect a reference grant something like:

```suggestion
func referenceGrantHasGatewayFrom(obj client.Object) bool {
```